src/java/org/apache/cassandra/streaming/StreamSession.java

```diff
@@ -694,8 +702,7 @@ public Future<?> onError(Throwable e)
             state(State.FAILED); // make sure subsequent error handling sees the session in a final state 
             channel.sendControlMessage(new SessionFailedMessage()).awaitUninterruptibly();
         }
-
-        return closeSession(State.FAILED);
+        return closeSession(State.FAILED, "Failed because of an unkown exception;\n" + Throwables.getStackTraceAsString(e));
```

Comment:
> The details of the exception aren't provided, only the stack trace which might lose info. We've already logged the main error details including the stacktrace, so a summary with just the name/cause may be more useful when debugging.

what details are you looking for?  What info is lost?

We may have logged the error, but that can be hard in debugging as we need to link 2 logs or a log with a vtable... IMO its best to include the exception so while debugging everything is in one place

Reply:
I think it's just printing the stack trace, not any message reason. Including the stacktrace in the message seems like it would generate a lot of output and be hard to read, but perhaps that's worth it.

Reply:
I believe it would be worth it because it doesn't affect readability by a large amount.

Reply:
marking this unresolved as I don't feel this feedback was addressed

Reply:
>  the exception isn't really unknown, it's just not EOF.

disagree, it really is unknown; the only expected (or known) exceptions are the EOF (closed) ones.

Reply:
I asked for the message to revert back, but the feedback I got in https://github.com/apache/cassandra/pull/2104#discussion_r1091003446 was that this request wanted to change it...